anime_downloader/sites/animesvostfr.py

```diff
@@ -0,0 +1,247 @@
+                p = v.img['src']
+                
+                lang = v.select_one('div.language').text.upper()
+                
+                #logger.debug( "results: {}".format(lang) )
+                
+                eps = v.select_one('span.mli-quality').text.upper()
+                
+                #logger.debug( "results: {}".format(eps) )
+                
+                if re.search('VOSTFR', lang, re.IGNORECASE):
+                    t += ' (VOSTFR '+eps+')'
+                elif re.search('VF', lang, re.IGNORECASE):
+                    t += ' (VF '+eps+')'
+                elif re.search('VO', lang, re.IGNORECASE):
+                    t += ' (VO '+eps+')'
+                else:
+                    t += ' (V? '+eps+')'
+                    
+                
+                search_result_info = SearchResult(title=t, url=u, poster=p)
+               
+                results.append(search_result_info)
+
+            
+            logger.debug( "results: {}".format(results) )
+            
+            return results
+        
+        def _scrape_episodes(self):
+            soup = helpers.soupify(helpers.get(self.url))
+
+            episode_links = None
+            results = [] 
+            
+            season = soup.find("div", id="seasonss")
+            
+            if season:
+                episode_links = season.select('div.les-title > a')
+                
+            
+            if episode_links:
+            
+                logger.debug( "episode_links: {}".format(episode_links) )
+                
+                #results = [a.get('href') for a in episode_links[::-1]]
+                results = [a.get('href') for a in episode_links]
+                
+                logger.debug( "results: {}".format(results) )
+            else:
+                results.append(self.url)
+                
+            
+            return results
+
+        def _scrape_metadata(self):
+            
+            logger.debug("_scrape_metadata url: %s", str(self.url) )
+            
+            soup = helpers.soupify(helpers.get(self.url, cf=True))
+     
+            h1 = soup.select_one('div.main-content h1')
+            logger.debug("_scrape_metadata H1: %s", str(h1))
+            
+            self.title = h1.text.strip()
+            
+
+class AnimesVOSTFREpisode(AnimeEpisode, sitename='animesvostfr'):
+        episodeId_url = 'https://www1.animesvostfr.net/api/episode'
+        stream_url = 'https://www1.animesvostfr.net/api/videos?episode_id'
+        anime_url = 'https://www.www1.animesvostfr.net/shows'
+        
+        link_stream_url = "https://www1.animesvostfr.net/ajax-get-link-stream/"
+        
+        SERVERS = [
+            'rapidvideo',
+            'photoss'
+        ]
+
+        def _get_sources(self):
+            
+            #version = self.config['version'] 
+            #server = self.config['server']
+            #fallback = self.config['fallback_servers']
```

Comment:
These are fine and should work, but you will need to add default configurations for them in config.py in the site config section (example: https://github.com/vn-ki/anime-downloader/blob/8b2654c9e7b13e853244e9d5903ecd08301b77a6/anime_downloader/config.py#L77-L81)

Reply:
Also you can determine user selected quality (i.e. with the `-q` flag) using `self.quality`.

Reply:
yes i comment this part because it is in my todo list :p

Reply:
Take a look at https://github.com/vn-ki/anime-downloader/blob/master/anime_downloader/sites/anime.py#L356-L378
for some help with returning server/version according to what the user selected in config. Note that this is optional. 
Example of usage: https://github.com/Blatzar/anime-downloader/blob/127c09fdb5075cba802ba2c3a0ade74a4f4ab751/anime_downloader/sites/animevibe.py#L45-L75